apps/examples/testcase/le_tc/network/itc_net_send.c

```diff
@@ -0,0 +1,181 @@
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific
+* language governing permissions and limitations under the License.
+*
+****************************************************************************/
+
+// @file itc_net_send.c
+// @brief ITC Test Case Example for send() API
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <pthread.h>
+
+#include "tc_internal.h"
+
+#define PORTNUM 1110
+
+static pthread_mutex_t g_lock = PTHREAD_MUTEX_INITIALIZER;
+static pthread_cond_t g_cond;
+
+/**
+* @testcase            :itc_net_send_n_fd
+* @brief               :used to transmit a message to another socket
+* @scenario            :sending invalid file descriptor
+* @apicovered          :send()
+* @precondition        :None
+* @postcondition       :None
+*/
+static void itc_net_send_n_fd(void)
+{
+	char *msg = "Hello World !\n";
+	int ret = send(-1, msg, strlen(msg), 0);
+
+	TC_ASSERT_EQ("send", ret, -1);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase            :itc_net_send_n_msg
+* @brief               :used to transmit a message to another socket
+* @scenario            :sending NULL msg
+* @apicovered          :send()
+* @precondition        :None
+* @postcondition       :None
+*/
+static void itc_net_send_n_msg(int server_socket)
+{
+	char *msg = NULL;
+	int ret = send(server_socket, msg, strlen(msg), 0);
+
+	TC_ASSERT_EQ("send", ret, -1);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @fn                   :server
+* @brief                :For client server scenario creating thread function
+* @scenario             :None
+* API's covered         :socket,bind,listen,close
+* Preconditions         :None
+* Postconditions        :None
+*/
+static void *server(void *args)
+{
+	struct sockaddr_in sa, client;
+	int ret;
+	int client_socket;
+	int addr_len = sizeof(client);
+	int server_socket = socket(PF_INET, SOCK_STREAM, IPPROTO_TCP);
+	if (server_socket == -1) {
+		printf("[%s]Failed to create socket\n", __func__);
+		return NULL;
+	}
+
+	memset(&sa, 0, sizeof(sa));
+
+	sa.sin_family = PF_INET;
+	sa.sin_port = htons(PORTNUM);
+	sa.sin_addr.s_addr = inet_addr("127.0.0.1");
+
+	ret = bind(server_socket, (struct sockaddr *)&sa, sizeof(sa));
+	if (ret == -1) {
+		printf("[%s]Failed to bind socket\n", __func__);
+		goto end;
+	}
+
+	ret = listen(server_socket, 2);
+	if (ret == -1) {
+		printf("[%s]Failed to listen socket\n", __func__);
+		goto end;
+	}
+
+	pthread_mutex_lock(&g_lock);
+	pthread_cond_signal(&g_cond);
+	pthread_mutex_unlock(&g_lock);
+
+	client_socket = accept(server_socket, (struct sockaddr *)&client, (socklen_t *)&addr_len);
+	if (client_socket < 0) {
+		goto end;
+	}
+
+	itc_net_send_n_fd();
+	itc_net_send_n_msg(server_socket);
```

Comment:
you can't send data by server_socket. you should use client_socket to send data to the client

Reply:
Fixed review comments